os.c

```diff
@@ -76,25 +76,23 @@ perf_os_handlesignal(int sig, void (*handler)(int))
 isc_result_t
 perf_os_waituntilwriteable(int fd, isc_int64_t timeout)
 {
-	fd_set write_fds;
-	struct timeval tv, *tvp;
+	struct pollfd write_fds[1];
+	int timeout_msec;
 	int n;
 
-	FD_ZERO(&write_fds);
-	FD_SET(fd, &write_fds);
+	write_fds[0].fd = fd;
+	write_fds[0].events = POLLOUT;
 	if (timeout < 0) {
-		tvp = NULL;
+		timeout_msec = 0;
 	} else {
-		tv.tv_sec = timeout / MILLION;
-		tv.tv_usec = timeout % MILLION;
-		tvp = &tv;
+		timeout_msec = timeout / THOUSAND;
 	}
-	n = select(fd + 1, NULL, &write_fds, NULL, tvp);
+	n = poll((struct pollfd*)&write_fds, 1, timeout_msec);
```

Comment:
I'm curious about why the cast is necessary. Surely

poll(write_fds, 1, timeout_msec);

should be OK?